### PR TITLE
55-badge-feedback

### PR DESCRIPTION
**Pull request by eriktks**
Refs #55 #102  

## Describe the changes made in this pull request

1. Compared generated fair-software badge with existing fair-software badge
2. Generated different messages for the user depending on this comparison 

## Instructions to review the pull request

1. Rebuild howfairis
2. Run `howfairis https://github.com/iomega/zenodo-upload` to check behavior for an upgraded badge
3. Run `howfairis https://github.com/fair-software/howfairis` to check reaction to a matching badge
4. Run `howfairis https://github.com/xtensor-stack/xtensor-fftw` to check output for a missing badge

### CODE REVIEW

File: howfairis/Compliance.py [hunk 1]
```diff
@@ -54,3 +65,11 @@ def __eq__(self, other):
...
             self.compliant_symbol == other.compliant_symbol and \
             self.noncompliant_symbol == other.noncompliant_symbol
+
+    def __gt__(self, other):
+        return \
+            (self.repository and not other.repository) or \
+            (self.license and not other.license) or \
+            (self.registry and not other.registry) or \
+            (self.citation and not other.citation) or \
+            (self.checklist and not other.checklist)
```
Comment (sverhoeven) [inline howfairis/Compliance.py:75]: Nicer to use count.
```suggestion
        return self.count(True) > other.count(True)
```

File: howfairis/Compliance.py [hunk 2]
```diff
@@ -45,6 +46,16 @@ def count(self, value):
...
 
+    def urldecode(self, string):
+        compliance_symbols = re.sub(" ", "", requests.utils.unquote(string))
+        if len(compliance_symbols) == 5:
+            self.repository = (compliance_symbols[0] == self.compliant_symbol)
+            self.license = (compliance_symbols[1] == self.compliant_symbol)
+            self.registry = (compliance_symbols[2] == self.compliant_symbol)
+            self.citation = (compliance_symbols[3] == self.compliant_symbol)
+            self.checklist = (compliance_symbols[4] == self.compliant_symbol)
+        return(self)
```
Comment (sverhoeven) [inline howfairis/Compliance.py:57]:
```suggestion
    @classmethod
    def urldecode(cls, string):
        compliance_symbols = re.sub(" ", "", requests.utils.unquote(string))
        if len(compliance_symbols) == 5:
            return  cls(
              repository = (compliance_symbols[0] == self.compliant_symbol),
              license = (compliance_symbols[1] == self.compliant_symbol),
              registry = (compliance_symbols[2] == self.compliant_symbol),
              citation = (compliance_symbols[3] == self.compliant_symbol),
              checklist = (compliance_symbols[4] == self.compliant_symbol)
            )
        return cls()
```

File: howfairis/cli.py [hunk 3]
```diff
@@ -87,14 +88,41 @@ def cli(url=None, branch=None, config_file=None, remote_config_file=None, path=N
...
+
+
+def get_fair_software_badge(text):
+    compliance = Compliance.Compliance()
+    url = "https://img.shields.io/badge/fair--software.eu"
+    url_location = text.find(url)
+    if url_location < 0:
+        return(False, compliance)
+    start_id = url_location+len(url)+1
+    return(True, compliance.urldecode(text[start_id:start_id+69]))
```
Comment (sverhoeven) [inline howfairis/cli.py:125]:
```suggestion
    url = "https://img.shields.io/badge/fair--software.eu"
    url_location = text.find(url)
    if url_location < 0:
        return(False, Compliance())
    start_id = url_location+len(url)+1
    return(True, Compliance.urldecode(text[start_id:start_id+69]))
```

File: howfairis/cli.py [hunk 4]
```diff
@@ -3,6 +3,7 @@
 import click
 from colorama import init as init_terminal_colors
 from howfairis import Checker
+from howfairis import Compliance
```
Comment (sverhoeven) [inline howfairis/cli.py:6]:
```suggestion
from howfairis.Compliance import Compliance
```

**Review by sverhoeven — APPROVED** [review]
The example commands worked as expected.

Switched from implemented object method to a class method to construct Compliance from a url.

**Comment by eriktks** [comment]
Thanks for the review and the updates!

**Pull request merged**